src/components/ModalPortal.js

```diff
@@ -84,6 +102,7 @@ export default class ModalPortal extends Component {
   }
 
   componentWillUnmount() {
+    this.removeModalInstance();
```

Comment:
I've moved the other ones, but in `componentWillUnmount` we aren't calling the close method at all, so the tests will fail.

We could call `close` method here, but I feel safer calling `removeModalInstance` directly so we don't run into potential issues if a `closeTimeoutMS` prop is set.